daktari/checks/onepassword.py

```diff
@@ -0,0 +1,68 @@
+import logging
+import re
+from typing import Optional
+import json
+
+from daktari.check import Check, CheckResult
+from daktari.command_utils import get_stdout
+from daktari.os import OS
+from pathlib import Path
+
+
+class OnePassInstalled(Check):
+    name = "onepass.installed"
+
+    def __init__(self, minimum_version: Optional[float] = None):
+        self.minimum_version = minimum_version
+        self.suggestions = {
+            OS.GENERIC: "Install OP (1Password CLI): "
+                        "https://support.1password.com/command-line-getting-started/#set-up-the-command-line-tool",
+        }
+
+    def check(self) -> CheckResult:
+        installed_version = get_op_version()
+        return self.validate_minimum_version("op", installed_version, self.minimum_version)
+
+
+version_pattern = re.compile("([0-9]+.[0-9]+)")
+
+
+def get_op_version() -> Optional[float]:
+    raw_version = get_stdout("op --version")
+    if raw_version:
+        match = version_pattern.search(raw_version)
+        if match:
+            version_string = match.group(1)
+            logging.debug(f"OP Version: {version_string}")
+            return float(version_string)
+    return None
+
+
+class OPAccountExists(Check):
+    depends_on = [OnePassInstalled]
+
+    def __init__(self, context_name: str):
+        self.context_name = context_name
+        self.name = f"op.accountExists.{context_name}"
+        self.suggestions = {
+            OS.GENERIC: f"<cmd>op signin {context_name}.onepassword.com <your-email-here></cmd>",
+        }
+
+    def check(self) -> CheckResult:
+
+        home = str(Path.home())
+        if OS.OS_X:
+            config_path = f"{home}/.op/config"
+        else:
+            config_path = f"{home}/.config/op/config"
+
+        op_config = json.loads(open(config_path).read())
```

Comment:
Might be worth a file existence check before we go ahead and try to open it? It blows up for me if it doesn't exist, rather than giving nice daktari output:

```
✅ [onepass.installed] op is installed
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/alex/daktari/daktari/__main__.py", line 35, in <module>
    sys.exit(main())
  File "/home/alex/daktari/daktari/__main__.py", line 29, in main
    all_passed = run_checks(config.checks)
  File "/home/alex/daktari/daktari/check_runner.py", line 13, in run_checks
    return CheckRunner(checks).run()
  File "/home/alex/daktari/daktari/check_runner.py", line 24, in run
    self.try_run_check(check)
  File "/home/alex/daktari/daktari/check_runner.py", line 32, in try_run_check
    self.run_check(check)
  File "/home/alex/daktari/daktari/check_runner.py", line 38, in run_check
    result = check.check()
  File "/home/alex/daktari/daktari/checks/onepassword.py", line 59, in check
    op_config = json.loads(open(config_path).read())
FileNotFoundError: [Errno 2] No such file or directory: '/home/alex/.op/config'
```

Reply:
Added!